feat(reactnative): Add source maps upload guide with Debug IDs (#8436)
getsentry/sentry-docs: src/platforms/react-native/manual-setup/hermes.mdx

  - /platforms/react-native/hermes/
---

Sentry React Native SDK supports `Hermes` engine out of the box.

## Source maps

Read our [React Native source maps documentation](/platforms/react-native/sourcemaps/) to learn how to upload source maps for your app.

## Throubleshooting

This section describes common issues when using Hermes and how to resolve them.

> ### Multiple ways of enabling source maps in Xcode

Reviewer: ```suggestion
### Multiple ways of Enabling Source Maps in Xcode
```

> If you use `EXTRA_PACKAGER_ARGS`, only the packager source map is needed. This is the default behavior if you used [`@sentry/wizard`](/platforms/react-native/#install). If you use [`SOURCEMAP_FILE`](https://reactnative.dev/docs/sourcemaps#enable-source-maps-on-ios), the combined Hermes soruc map is necessary.

Reviewer: ```suggestion
If you use `EXTRA_PACKAGER_ARGS`, you only need the packager source map. This is the default behavior if you used [`@sentry/wizard`](/platforms/react-native/#install). If you use [`SOURCEMAP_FILE`](https://reactnative.dev/docs/sourcemaps#enable-source-maps-on-ios), the combined Hermes source map is necessary.
```

> ### Check steps of RN tooling

Reviewer: ```suggestion
### Check RN Tooling Steps
```

> Our [React Native source maps documentation](/platforms/react-native/sourcemaps/) is based on [`react-native-xcode.sh`](https://github.com/facebook/react-native/blob/0.71-stable/scripts/react-native-xcode.sh) for iOS and on [`BundleHermesCTask.kt`](https://github.com/facebook/react-native/blob/0.71-stable/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/BundleHermesCTask.kt) for Android. If you are having issues with the manually generated artifacts check the steps of your React Native tooling using the meantioned links.

Reviewer: ```suggestion
Our React Native [Source Maps](/platforms/react-native/sourcemaps/) documentation is based on [`react-native-xcode.sh`](https://github.com/facebook/react-native/blob/0.71-stable/scripts/react-native-xcode.sh) for iOS and on [`BundleHermesCTask.kt`](https://github.com/facebook/react-native/blob/0.71-stable/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/BundleHermesCTask.kt) for Android. If you are having issues with the manually generated artifacts check the steps of your React Native tooling using the previous links.
```

<Note>

React Native [`0.70`](https://github.com/facebook/react-native/blob/e3a5fbe72f966b27b967192317d7072db52d1c8c/scripts/react-native-xcode.sh#L141) and CodePush Update minifies the packager bundle by default.

</Note>

<Note>

[`BundleHermesCTask.kt`](https://github.com/facebook/react-native/blob/0.71-stable/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/BundleHermesCTask.kt) is only available from version `0.71` on. For previous versions, use [`react.gradle`](https://github.com/facebook/react-native/blob/0.70-stable/react.gradle).

</Note>

For more details see [this issue](https://github.com/facebook/react-native/issues/34043) in the React Native repository.

> ### Don't combine Hermes and RAM bundles

Reviewer: ```suggestion
### Don't Combine Hermes and RAM Bundles
```

If you are using Hermes, you should not have the [RAM bundles](/platforms/react-native/manual-setup/ram-bundles/) feature enabled. In Hermes, when loading the bytecode, `mmap` ensures that the entire file is not loaded. Using Hermes with RAM bundles might lead to issues, because those mechanisms are not compatible with each other.

### Minimum Required Versions

Sentry added support for `react-native` builds that use the `hermes` engine, which required changes to the Sentry SDK, `sentry-cli` as well as Sentry itself.

> Sentry customers using the SaaS product (sentry.io) will need to update the SDK, and `sentry-cli`.

Reviewer: We always link when we write out sentry.io

```suggestion
Sentry customers using the SaaS product ([sentry.io](https://sentry.io/)) will need to update the SDK and `sentry-cli`.
```
